components/shared_code/pyproject.toml

```diff
@@ -25,7 +25,7 @@ tools = [
     "black==23.9.1",
     "mypy==1.5.1",
     "pip-audit==2.6.1",
-    "pydantic==1.10.12",
+    "pydantic==2.3.0",
```

Comment:
Because `ci/quality.sh` needs to inject Pydantic into the mypy pipx installation and the `spec` functions looks in the tools list for the version number of tools.